web/app/js/components/LineGraph.jsx

```diff
@@ -133,6 +141,39 @@ export default class LineGraph extends React.Component {
     }
   }
 
+  flashLatestDataPoint() {
+    if (!this.props.flashLastDatapoint) {
+      return;
+    }
+
+    let circle = this.svg.select("circle");
+    if (_.isEmpty(this.props.data)) {
+      circle.attr("flashing", "off").interrupt().style("opacity", 0);
+    } else {
+      if (circle.attr("flashing") === "off") {
+        circle
+          .attr("flashing", "on")
+          .transition()
+          .on("start", function repeat() {
+            d3.active(this)
+              .transition()
+              .duration(1000)
+              .style("opacity", 0.6)
+              .transition()
+              .duration(1000)
+              .style("opacity", 0)
+              .transition()
+              .on("start", repeat);
+          });
+      }
+
+      let circleData = _.last(this.props.data);
```

Comment:
oops sorry I didn't catch this before, but I think you want this inside the else {}
otherwise you'll try to do a _.last whether data is empty or not

Reply:
oh whoops nvm looks like it /is/ in the else

Reply:
:) i've moved `let circleData` up to the beginning of the else statement to hopefully make it clearer upon first look